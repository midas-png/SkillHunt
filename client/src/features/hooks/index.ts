export { useOnScreen } from './useOnScreen';
export { useToggle } from './useToggle';
export { useAppDispatch, useAppSelector } from './useRedux';
