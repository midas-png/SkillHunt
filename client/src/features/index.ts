export { defaultTheme } from './theme';
export { renderWithTheme } from './theme/render-methods';
