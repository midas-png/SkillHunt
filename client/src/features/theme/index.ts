import { DefaultTheme } from 'styled-components';

export const defaultTheme: DefaultTheme = {
  fontFamily: {
    primary: 'Work Sans',
    secondary: 'Roboto',
  },
  palette: {
    primary: '#fff',
    secondary: '#000',
    tertiary: '#eafa35',
    placeholder: '#555',
  },
};
