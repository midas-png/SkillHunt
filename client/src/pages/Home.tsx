import { Hero } from 'components';

export const Home = () => <Hero />;
