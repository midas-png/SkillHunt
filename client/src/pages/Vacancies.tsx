import { FC } from 'react';
import { VacanciesHead } from 'components';

export const Vacancies: FC = () => (
  <>
    <VacanciesHead />
  </>
);
