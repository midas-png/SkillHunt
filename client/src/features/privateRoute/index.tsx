import { FC } from 'react';
import { Navigate, Outlet } from 'react-router-dom';
import { IProps } from './props';
import { useAppSelector } from 'features';

export const PrivateRoute: FC<IProps> = ({ authRoute, redirectPath }) => {
  const { isAuth } = useAppSelector((state) => state.userReducer);
  const auth = authRoute ? isAuth : !isAuth;

  if (auth) {
    return <Outlet />;
  }

  return <Navigate to={redirectPath ? redirectPath : '/signin'} />;
};
