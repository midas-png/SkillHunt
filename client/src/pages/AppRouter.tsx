import { Route, Routes } from 'react-router-dom';
import { Home } from './Home';
import { PrivateRoute } from 'features';

export const AppRouter = () => {
  return (
    <Routes>
      <Route element={<PrivateRoute />}>
        <Route path='/' element={<Home />} />
      </Route>
    </Routes>
  );
};
