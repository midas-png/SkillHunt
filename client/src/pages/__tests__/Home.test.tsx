import { screen } from '@testing-library/react';
import { renderWithTheme } from 'features';
import { Home } from 'pages/Home';

test('Home page render', () => {
  renderWithTheme(<Home />);
  const title1 = screen.getByText(/^Land a job$/i);
  const title2 = screen.getByText(/^the easy way$/i);
  const subtitle = screen.getByText(
    /The best place to discover & apply to the coolest start up jobs, without the black box./i
  );
  const button = screen.getByText(/^explore$/i);

  expect(title1).toBeInTheDocument;
  expect(title2).toBeInTheDocument;
  expect(subtitle).toBeInTheDocument;
  expect(button).toBeInTheDocument;
});
