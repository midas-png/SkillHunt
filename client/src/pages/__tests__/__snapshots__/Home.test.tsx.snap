// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home page render Home page snapshot tests 1`] = `
<span
  class="sc-qRumB cBNnCH"
  font-size="120"
>
  Land a job
</span>
`;

exports[`Home page render Home page snapshot tests 2`] = `
<span
  class="sc-qRumB cBNnCH"
  font-size="120"
>
  the easy way
</span>
`;

exports[`Home page render Home page snapshot tests 3`] = `
<span
  class="sc-jTrPJq kSlZxv"
>
  The best place to discover & apply to the coolest start up jobs, without the black box.
</span>
`;

exports[`Home page render Home page snapshot tests 4`] = `
<button
  class="sc-iAEyYk cxsxSs"
>
  Explore
</button>
`;
